code/modules/reagents/reagent_dispenser.dm

```diff
@@ -1,196 +1,340 @@
+	update_icon()
 
-/obj/structure/reagent_dispensers/examine(mob/user)
-	..()
+/obj/structure/reagent_dispensers/update_icon()
+	overlays.Cut()
+	if(icon_state in bypassUIicons) return // can't do overlay fun things on stuff like peppertank/virus tank/watercooler
 	if(reagents.total_volume)
-		user << "<span class='notice'>It has [reagents.total_volume] units left.</span>"
-	else
-		user << "<span class='danger'>It's empty.</span>"
-
+		switch(reagents.get_master_reagent_id())//is master chem weldingfuel?change the iconstate into weldingfuel tank,same for water
+			if("welding_fuel")
+				name = "fueltank"
+				desc = "It can hold large amount of chemicals. Mainly used to hold fuel."
+				icon_state = "weldtank"
+			if("water")
+				name = "watertank"
+				desc = "It can hold large amount of chemicals. Mainly used to hold water."
+				icon_state = "watertank"
+			else
+				name = "storage tank"
+				desc = "It can hold large amount of chemicals."
+				icon_state = "chemtank"
+		var/image/filling = image('icons/obj/chemtank.dmi', src, "tankfilling1")
+		var/percent = round((reagents.total_volume / reagents.maximum_volume) * 100)
+		switch(percent)
+			if(0 to 25)			filling.icon_state = "tankfilling1"
+			if(25 to 50)		filling.icon_state = "tankfilling2"
+			if(50 to 75)		filling.icon_state = "tankfilling3"
+			if(75 to INFINITY)	filling.icon_state = "tankfilling4"
+		filling.color = mix_color_from_reagents(reagents.reagent_list)
+		overlays += filling
+	if(!(flags & INJECTONLY))//here we check the lid!we want it to appear if it's closed.
+		var/image/lid = image('icons/obj/chemtank.dmi', src, "tanklid")
+		overlays += lid
 
-/obj/structure/reagent_dispensers/proc/boom()
-	visible_message("<span class='danger'>\The [src] ruptures!</span>")
-	chem_splash(loc, 5, list(reagents))
+/obj/structure/reagent_dispensers/proc/boom(atom/A) // detonate and explode were already taken and i hate two procs with the same name
+	if(A && istype(A, /mob))
+		var/mob/user = A
+		message_admins("[key_name_admin(user)] triggered a chemtank explosion at [src ? "[x],[y],[z]" : "Carbonhell fucked up again, whine at him"] - <A HREF='?_src_=holder;adminplayerobservecoodjump=1;X=[x];Y=[y];Z=[z]'>JMP</a>.")
+		log_game("[key_name(user)] triggered a chemtank explosion at [src ? "[x],[y],[z]" : "Carbonhell fucked up again, whine at him"].")
+		user.attack_log += text("\[[time_stamp()]\] <font color='red'>Has detonated a chem tank @ [src ? "[x],[y],[z]" : "UNKNOWN LOCATION"]</font>")
+	if(reagents && reagents.reagent_list.len && !exploded)
+		reagents.chem_temp = 1000
+		reagents.handle_reactions()
+		if(!src) return 1// safety check,explosion could've deleted it!
+		exploded = 1
+		var/volumepereffect = reagents.total_volume / 9
+		var/list/effectlist = list()
+		visible_message("<span class='danger'>[src] ruptures, spraying its contents everywhere!</span>")
+		spawn(-1)
+			for(var/i in 1 to 11) // i will be later used for direction. i'm a smart man am i not?admit it fucker

```

Comment:
I'm quite sure there is a proc that gets all possible directions and/or execute something for all directions